build/__tools__/run-tests-with-database.sh

```diff
@@ -0,0 +1,189 @@
+if [[ $PY3_VERSION -lt 5 ]]; then
+    die activate python3 first
+fi
+
+SCRIPTDIR="$(dirname $(readlink --canonicalize ${BASH_SOURCE}))"
+SRCDIR="$(realpath ${SCRIPTDIR}/../..)"
+
+
+
+: "${PDO_HOME:-$(die Missing environment variable PDO_HOME)}"
+: "${PDO_LEDGER_URL:-$(die Missing environment variable PDO_LEDGER_URL)}"
+
+# check for existing enclave and provisioning services
+pgrep eservice
+if [ $? == 0 ] ; then
+    die existing enclave services detected, please shutdown
+fi
+
+pgrep pservice
+if [ $? == 0 ] ; then
+    die existing provisioning services detected, please shutdown
+fi
+
+SAVE_FILE=$(mktemp /tmp/pdo-test.XXXXXXXXX)
+ESERVICE_DB=${PDO_HOME}/data/eservice-db.json
+
+declare -i NUM_SERVICES=5 # must be at least 3 for pconntract update test to work
+function cleanup {
+    read -p "Enter any key to shutdown services."
+    yell "shutdown services"
+    ${PDO_HOME}/bin/ps-stop.sh --count ${NUM_SERVICES} > /dev/null
+    ${PDO_HOME}/bin/es-stop.sh --count ${NUM_SERVICES} > /dev/null
+    ${PDO_HOME}/bin/ss-stop.sh --count ${NUM_SERVICES} > /dev/null
+    rm -f ${SAVE_FILE}
+    rm -r ${ESERVICE_DB}
+}
+
+trap cleanup EXIT
+
+
+# -----------------------------------------------------------------
+yell start enclave and provisioning services
+# -----------------------------------------------------------------
+try ${PDO_HOME}/bin/ss-start.sh --count ${NUM_SERVICES} > /dev/null
+try ${PDO_HOME}/bin/ps-start.sh --count ${NUM_SERVICES} --ledger ${PDO_LEDGER_URL} --clean > /dev/null
+try ${PDO_HOME}/bin/es-start.sh --count ${NUM_SERVICES} --ledger ${PDO_LEDGER_URL} --clean > /dev/null
+
+cd ${SRCDIR}/build 
+
+#-----------------------------------------
+yell test the eservice database 
+#------------------------------------------
+say run module level tests for database manager
+cd ${SRCDIR}/python/pdo/test
+try python servicedb.py --logfile $PDO_HOME/logs/client.log --loglevel info \
+    --eservice-db $PDO_HOME/data/db-test.json --url http://localhost:7101 http://localhost:7102 http://localhost:7103
+try rm $PDO_HOME/data/db-test.json
+
+cd ${SRCDIR}/build
+
+say create the eservice database using database CLI 
+try pdo-create-eservicedb --logfile $PDO_HOME/logs/client.log --loglevel info \
+    --eservice-db ${ESERVICE_DB} --eservice-url http://localhost:7101 --eservice-name e1
+# add more entires
+for v in $(seq 2 ${NUM_SERVICES}) ; do
+    try pdo-add-to-eservicedb --logfile $PDO_HOME/logs/client.log --loglevel info \
+        --eservice-db ${ESERVICE_DB} --eservice-url http://localhost:710${v} --eservice-name e${v}
+done
+
+say run various pdo scripts - test-request, test-contract, create, update, shell - using database
+
+
+try pdo-test-request --no-ledger   \
+    --eservice-name e1 --logfile $PDO_HOME/logs/client.log --loglevel info --eservice-db ${ESERVICE_DB}
+
+try pdo-test-contract --no-ledger --contract integer-key \
+    --eservice-name e2 --logfile $PDO_HOME/logs/client.log --loglevel info --eservice-db ${ESERVICE_DB}
+
```

Comment:
any particular reason why you aren't defaulting the eservice db in either the TOML file or in the command parameters? Seems like for these tests, we should just assume the defaults.

Reply:
Yes, i agree, i will add a default db file name to toml.